go/consensus/api/mode/mode.go

```diff
@@ -0,0 +1,51 @@
+// Package mode provides the implementation agnostic consensus mode type.
+package mode
+
+import "fmt"
+
+// Mode is the consensus node mode.
+type Mode string
+
+const (
+	// ModeFull is the name of the full node consensus mode.
+	ModeFull Mode = "full"
+	// ModeArchive is the name of the archive node consensus mode.
+	ModeArchive Mode = "archive"
```

Comment:
Since we are here, why is this needed in addition to the global node mode? Could it be removed?

Reply:
I've removed this, but it appears that the archive API test wants to have a node that's both a compute and an archive node...  What is the use case for this?  Do we need to keep this functionality or is it OK to remove?

Reply:
Huh, which test wants that? It doesn't make any sense.

Reply:
I only see a compute node that is _turned into_ an archive node. So it is not both at the same time.

Reply:
The test code seems to have a notion of a "consensus archive node" and a "runtime archive node" -- but an archive node as defined in the main node mode cannot have runtimes.  So somebody added this notion of a "runtime archive node", but why?

Reply:
Every archive node should be a runtime archive node if it has any runtimes defined. cc @ptrus

Reply:
I think that this duplication was done in: https://github.com/oasisprotocol/oasis-core/pull/4964

Since originally, I see only the mode defined under consensus (https://github.com/oasisprotocol/oasis-core/pull/4571).

Feel free to assign me on the issue for fixing/cleaning this up if noone else will.

Reply:
If it's not too bad we should try to fix it in this PR.

Reply:
It's OK, I've already fixed this in this PR :)